File: src/console/command/parameter/def/FloatParameter.php
<?php

namespace pocketcloud\cloud\console\command\parameter\def;

use pocketcloud\cloud\console\command\parameter\exception\ArgumentParseException;
use pocketcloud\cloud\console\command\parameter\CommandParameter;

readonly class FloatParameter extends CommandParameter {

    public function parseValue(string $input): int {
        if (is_numeric($input)) return floatval($input);
        return throw new ArgumentParseException();
    }

    public function getType(): string {
        return "float";
    }
}

File: src/main/java/de/pocketcloud/cloud/console/command/parameter/def/FloatParameter.java
package de.pocketcloud.cloud.console.command.parameter.def;

import de.pocketcloud.cloud.console.command.parameter.BaseCommandParameter;
import de.pocketcloud.cloud.console.command.exception.ArgumentParseException;

public class FloatParameter extends BaseCommandParameter {

    public FloatParameter(String name, boolean optional, String customErrorMessage) {
        super(name, optional, customErrorMessage);
    }

    public FloatParameter(String name, boolean optional) {
        super(name, optional);
    }

    @Override
    public Float parseValue(String input) throws ArgumentParseException {
        try {
            return Float.parseFloat(input);
        } catch (NumberFormatException e) {
            throw new ArgumentParseException();
        }
    }

    @Override
    public String getType() {
        return "float";
    }
}